src/main/java/practice/StreamPractice.java

```diff
@@ -50,7 +77,9 @@ public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toA
      */
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> CheckedPerson.check(p, fromAge, femaleToAge, maleToAge))
+                .collect(Collectors.toList());
```

Comment:
It's not a good idea to use inner class in this place. 
"filter()" consumes Predicate. It's better to implement Predicate here.                
```suggestion
       Predicate <Person> personPredicate = ... ;
                return peopleList
                    .stream()
                    .filter(personPredicate)
                    .collect(Collectors.toList());
```